examples/pytorch/pytorch_checkpoint.py

```diff
@@ -0,0 +1,194 @@
+"""
+Optuna example that optimizes multi-layer perceptrons using PyTorch with checkpoint.
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+PyTorch and MNIST. We optimize the neural network architecture as well as the optimizer
+configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
+subset of it.
+
+Even if the process where the trial is running is killed for some reason, you can restart from
+previous saved checkpoint using heartbeat.
+
+"""
```

Comment:
It'd be better to give example commands to run&kill this example first and restart hyperparameter optimization because this example is different from others in that this example will be once killed and resumed.
```suggestion
"""
Optuna example that optimizes multi-layer perceptrons using PyTorch with checkpoint.

In this example, we optimize the validation accuracy of hand-written digit recognition using
PyTorch and MNIST. We optimize the neural network architecture as well as the optimizer
configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
subset of it.

Even if the process where the trial is running is killed for some reason, you can restart from
previous saved checkpoint using heartbeat.

    $ timeout 20 python examples/pytorch/pytorch_checkpoint.py
    $ python examples/pytorch/pytorch_checkpoint.py
"""
```